### PR TITLE
Pipeline should test the examples

### LINKED ISSUE #98: Add documentation into mvg CI pipeline
COnsider env varaiables, check python-dotenv

### PR DESCRIPTION
**Pull request by vnadhan**
# Description

- New task added in `ci-checks` to validate examples on a new PR / push to the master branch
- Development requirements file split up into `requirements_dev` and `requirements_docs`
- Renamed filename of every example to include a number to define the order of execution
- Created a shell script to validate the examples using package called `papermill`
- Few fixes done to the examples (Refer https://github.com/vikinganalytics/mvg/commit/f1c54274dbd89a24645a98dfbe63d2cef6ff1a41)

New dependencies:
- `requirements_docs`
   - `tqdm`, `papermill`
   - removed dependency `notebook`
- refactored  `requirements_dev` 

Fixes #98 

## Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue -> bump patch)
- [ ] New feature (non-breaking change which adds functionality -> bump minor version)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected -> bump major version)
- [x] Documentation Update

## Checklist

- [ ] I have added tests that prove that my fix/feature works
- [ ] Linters pass locally and I have followed PEP8 code style
- [ ] New and existing tests pass locally
- [x] I have updated the documentation if needed
- [ ] I have commented hard-to-understand areas in the code

## Requirements

- [ ] I have updated the MVG version

### CODE REVIEW

File: regenerate-examples.sh [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+# Cleanup
+do_cleanup() {
+    echo "Performing cleanup of temporary resources"
+    rm -rf ./va-data-charlie
+}
+
+# Execute notebooks containing examples
+exec_examples() {
+    echo "Running notebooks containing examples"
+    for file in ./docs/source/content/examples/*.ipynb; do
```
Comment (tuix) [inline regenerate-examples.sh:17]: Is it always guaranteed that resulting files are sorted by name?

**Review by tuix — CHANGES_REQUESTED** [review]
It is better to use 2 digits prefixes for jupyter notebook file names. If we pass 9 examples sorting becomes an issue.

File: regenerate-examples.sh (hunk 1 above)
Comment (vnadhan) [reply regenerate-examples.sh]: https://github.com/vikinganalytics/mvg/pull/128/commits/8f150e5348dc597909c7c13b5f51e1284599b744 Fixed using the natural sort (ascending order) of `ls`.

**Comment by vnadhan** [comment]
> It is better to use 2 digits prefixes for jupyter notebook file names. If we pass 9 examples sorting becomes an issue.

I believe this will not be an issue if we are use the natural sorting behavior of the `ls` command.

**Review by tuix — APPROVED** [review]

**Pull request merged**